packages/block-library/src/embed/index.js

```diff
@@ -136,6 +137,58 @@ export function getEmbedEdit( title, icon ) {
 			return false;
 		}
 
+		/**
+		 * Sets the appropriate CSS class to enforce an aspect ratio when the embed is resized
+		 * if the HTML has an iframe with width and height set.
+		 *
+		 * @param {string} html The preview HTML that possibly contains an iframe with width and height set.
+		 */
+		maybeSetAspectRatioClassName( html ) {
+			const previewDom = document.createElement( 'div' );
+			previewDom.innerHTML = html;
```

Comment:
I am very wary of the security implications of this line. This will allow JavaScript in the preview markup to be evaluated, even though we're never including it in the DOM. As a demonstration, paste the following into your console while viewing the editor:

```js
document.createElement( 'div' ).innerHTML = '<img src=/ onerror=\'alert("haxd")\'>'
```

Now it's a question as to whether we consider embed preview markup to be "safe". Given that the Sandbox component exists, I'm operating on the assumption that the answer is **no**. Therefore, this is a security vulnerability if it comes to be released.

Reply:
Changed this to use a regex in https://github.com/WordPress/gutenberg/pull/9770

Reply:
Another simpler option might be `document.implementation.createHTMLDocument` as a "sandbox" of sorts.

Example: https://github.com/aduth/hpq/blob/f17b3fdc9c5692e9f676f94c33a003d191c81bd6/src/index.js#L21-L23